semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbTaskListener.java

```diff
@@ -44,7 +45,21 @@ public SemanticdbTaskListener(
   }
 
   @Override
-  public void started(TaskEvent e) {}
+  public void started(TaskEvent e) {
+    // Upon first encounter with a file (before any other tasks are run)
+    // we remove the semanticdb file for this source file to ensure
+    // stale data doesn't cause problems
```

Comment:
Now that we read previous state, we shouldn't expect the folder to be clean – without doing so I was getting very confusing results when running snapshots.